### PR TITLE
Signer interface update

### PR DESCRIPTION
**Pull request by woomurf**
- Add sendTransaction, sendTransactionBatch, sendSignedTransaction to signer interface.

Issue: #144

### CODE REVIEW

File: src/signer/ain-wallet-signer.ts [hunk 1]
```diff
@@ -36,4 +37,22 @@ export class AinWalletSigner implements Signer {
...
+   * Signs and sends a transaction to the network
+   * @param {TransactionInput} transactionObject
+   * @param {boolean} isDryrun - dryrun option.
+   * @return {Promise<any>}
+   */
+  sendTransaction(transactionObject: TransactionInput) {
+    return this.ainetwork.sendTransaction(transactionObject);
+  }
+
+  sendTransactionBatch(transactionObjects: TransactionInput[]): Promise<any> {
```
Comment (woomurf) [inline src/signer/ain-wallet-signer.ts:51]: Currnetly, Ain wallet does not support these functions.

File: src/signer/default-signer.ts [hunk 2]
```diff
@@ -0,0 +1,147 @@
...
+  }
+
+  /**
+   * Sends a signed transaction to the network
+   * @param {string} signature
+   * @param {TransactionBody} txBody
+   * @param {boolean} isDryrun - dryrun option.
+   * @return {Promise<any>}
+   */
+  async sendSignedTransaction(signature: string, txBody: TransactionBody, isDryrun: boolean = false): Promise<any> {
```
Comment (woomurf) [inline src/signer/default-signer.ts:99]: I moved this function to reduce code duplicated. Please comment about this. (But, this function does not require sign)

File: src/signer/default-signer.ts [hunk 3]
```diff
@@ -0,0 +1,147 @@
...
+   * If an address is not given, the default address of the wallet is used.
+   * @param {string} message - The message to sign.
+   * @param {string} address - The address of the account to sign the message with.
+   */
+  signMessage(message: string, address?: string): Promise<string> | string {
+    return this.wallet.sign(message, address);
+  }
+
+  /**
+   * Signs and sends a transaction to the network
```
Comment (platfowner) [inline src/signer/default-signer.ts:45]: minor) period mark (.) at the end?

File: src/signer/default-signer.ts [hunk 4]
```diff
@@ -0,0 +1,147 @@
...
+  }
+
+  /**
+   * Sends signed transactions to the network.
+   * @param {TransactionInput[]} transactionObjects
+   */
+  async sendTransactionBatch(transactionObjects: TransactionInput[]): Promise<any> {
+    let promises: Promise<any>[] = [];
+    for (let tx of transactionObjects) {
+      promises.push(this.buildTransactionBody(tx).then(async (txBody) => {
```
Comment (platfowner) [inline src/signer/default-signer.ts:63]: question) Is this the only place where buildTransactionBody() is used?

**Review by platfowner — APPROVED** [review]
LGTM with some minor comments.
Thanks!

File: src/signer/default-signer.ts (hunk 4 above)
Comment (woomurf) [reply src/signer/default-signer.ts:63]: In this code, `buildTransactionBody` is used only in `sendTransaction` and `sendTransactionBatch`.

Comment (platfowner) [reply src/signer/default-signer.ts:63]: Thanks for the confirmation!

**Comment by woomurf** [comment]
Thanks for the review! Merging now!

**Pull request merged**